slick/src/main/scala/slick/util/ReadAheadIterator.scala

```diff
@@ -49,3 +43,11 @@ trait ReadAheadIterator[+T] extends BufferedIterator[T] {
     } else throw new NoSuchElementException("next on empty iterator");
   }
 }
+
+object ReadAheadIterator {
+
+  /** Feature implemented in Scala library 2.12 this maintains functionality for 2.11 */
+  implicit class headOptionReverseCompatibility[T](readAheadIterator: ReadAheadIterator[T]){
+    def headOption : Option[T] = if (readAheadIterator.hasNext) Some(readAheadIterator.head) else None
```

Comment:
This looks like it has different behaviour from the old method in at least one way, since it doesn't throw an exception if the iterator is empty. Not that I don't prefer both the syntax and semantics of what you've done here - but if this is for backwards compatibility shouldn't it guarantee the same outputs?